tests/phpunit/tests/blocks/render.php

```diff
@@ -286,6 +286,56 @@ public function test_dynamic_block_rendering() {
 		);
 	}
 
+	/**
+	 * @ticket 62114
+	 */
+	public function test_dynamic_block_with_default_attributes() {
+		$settings = array(
+			'attributes'      => array(
+				'content'         => array(
+					'type'    => 'string',
+					'default' => 'Default content.',
+				),
+				'align'           => array(
+					'type'    => 'string',
+					'default' => 'full',
+				),
+				'backgroundColor' => array(
+					'type'    => 'string',
+					'default' => 'blueberry-1',
+				),
+				'textColor'       => array(
+					'type'    => 'string',
+					'default' => 'white',
+				),
+			),
+			'supports'        => array(
+				'align' => array( 'wide', 'full' ),
+				'color' => array(
+					'background' => true,
+					'text'       => true,
+				),
+			),
+			'render_callback' => function ( $attributes ) {
+				return '<p ' . get_block_wrapper_attributes() . '>' . $attributes['content'] . '</p>';
+			},
+		);
+		register_block_type( 'core/dynamic', $settings );
+
+		$post_content =
+			'before' .
+			'<!-- wp:core/dynamic --><!-- /wp:core/dynamic -->' .
+			'after';
+
+		$updated_post_content = do_blocks( $post_content );
+		$this->assertSame(
+			$updated_post_content,
+			'before' .
+			'<p class="alignfull wp-block-dynamic has-text-color has-white-color has-background has-blueberry-1-background-color">Default content.</p>' .
```

Comment:
Before/after of this change:

```diff
-before<p class="wp-block-dynamic">Default content.</p>after
+before<p class="alignfull wp-block-dynamic has-text-color has-white-color has-background has-blueberry-1-background-color">Default content.</p>after
```

Reply:
I copied an existing test. I guess it’s also fine to remove before & after strings as they don’t add too much value here.

Reply:
I'm not too worried about what's in the test. My comment was to capture how the changes in the PR changes the behavior of the new test, and that it would fail without the changes in this PR.